### PR TITLE
Configure performance test to include full path

### PR DESCRIPTION
**Pull request by koparasy**
## Proposed changes

When creating performance tests, the input files contained relative paths. Therefore, requiring the execution to happen from pre-defined directories. This branch changes that behavior, it describes input files through their absolute path.

## What type(s) of changes does this code introduce?
- Testing changes (e.g. new unit/integration/performance tests)

### Does this introduce a breaking change?

- No

## Checklist

_Update the following with a yes where the items apply. If you're unsure about any of them, don't hesitate to ask.  This is
simply a reminder of what we are going to look for before merging your code._

- No. This PR is up to date with current the current state of 'develop'
- Yes Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

**Comment by ye-luo** [comment]
Here is in my build directory.
```
build_rtx3060_clang_offload_cuda_real_MP$ ls tests/performance/NiO
$ ls tests/performance/NiO/
CMakeFiles                          dmc-a16-e192-SM1-cpu_driver         dmc-a4-e48-DU8-cpu_driver         dmc-a8-e96-SM1-cpu_driver                   NiO-fcc-supertwist111-supershift000-S32.h5
cmake_install.cmake                 dmc-a256-e3072-DU64-batched_driver  dmc-a4-e48-DU8-J3-cpu_driver      dmc-a96-e1152-DU32-batched_driver           NiO-fcc-supertwist111-supershift000-S4.h5
CTestTestfile.cmake                 dmc-a256-e3072-DU64-cpu_driver      dmc-a4-e48-SM1-cpu_driver         dmc-a96-e1152-DU32-cpu_driver               NiO-fcc-supertwist111-supershift000-S64.h5
dmc-a128-e1536-DU32-batched_driver  dmc-a256-e3072-DU64-J3-cpu_driver   dmc-a64-e768-DU32-batched_driver  dmc-a96-e1152-DU32-J3-cpu_driver            NiO-fcc-supertwist111-supershift000-S8.h5
dmc-a128-e1536-DU32-cpu_driver      dmc-a256-e3072-SM1-cpu_driver       dmc-a64-e768-DU32-cpu_driver      dmc-a96-e1152-SM1-cpu_driver                Ni.opt.xml
dmc-a128-e1536-DU32-J3-cpu_driver   dmc-a32-e384-DU32-batched_driver    dmc-a64-e768-DU32-J3-cpu_driver   Makefile                                    O.xml
dmc-a128-e1536-SM1-cpu_driver       dmc-a32-e384-DU32-cpu_driver        dmc-a64-e768-SM1-cpu_driver       NiO-fcc-supertwist111-supershift000-S16.h5  process_perf.py
dmc-a16-e192-DU16-batched_driver    dmc-a32-e384-DU32-J3-cpu_driver     dmc-a8-e96-DU16-batched_driver    NiO-fcc-supertwist111-supershift000-S1.h5
dmc-a16-e192-DU16-cpu_driver        dmc-a32-e384-SM1-cpu_driver         dmc-a8-e96-DU16-cpu_driver        NiO-fcc-supertwist111-supershift000-S24.h5
dmc-a16-e192-DU16-J3-cpu_driver     dmc-a4-e48-DU8-batched_driver       dmc-a8-e96-DU16-J3-cpu_driver     NiO-fcc-supertwist111-supershift000-S2.h5
```
*.h5, O.xml and Ni.opt.xml are all softlinks with absolute path and I can move the NiO folder to any desired place and run.
Optionally, set `QMC_SYMLINK_TEST_FILES=OFF` triggers copying h5 and xml files and NiO folder is self-contained for running.

**Comment by markdewing** [comment]
It sounds like you want to run the performance tests from a directory that is not the one with the input file.  Could you say more about the use-case for running the tests this way?

**Comment by koparasy** [comment]
I am auto-tuning the performance tests. The process stores large files in the directory I am executing the test from. Using the HEAD input files, I either need to copy files around and symlink everything correctly. 

IMO, it is more natural for the performance test to be agnostic to the running directory, and all the required information can be stored in the initial input file. Such a design approach transforms the performance test as a stand-alone case with no extra requirements.

File: tests/performance/NiO/CMakeLists.txt [hunk 1]
```diff
@@ -22,8 +22,15 @@ function(
   endif()
 
   set(PPDIR "${qmcpack_SOURCE_DIR}/tests/performance/NiO")
-
-  execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${SDIR} ${WDIR})
+  set(PNBDIR "${qmcpack_BINARY_DIR}/tests/performance/NiO/")
+
+  #  execute_process(COMMAND ${CMAKE_COMMAND} -E copy_directory ${SDIR} ${WDIR})
+  execute_process(COMMAND ${CMAKE_COMMAND} -E make_directory ${WDIR})
```
Comment (quantumsteve) [inline tests/performance/NiO/CMakeLists.txt:28]:
```suggestion
  file(MAKE_DIRECTORY ${WDIR})
```
https://cmake.org/cmake/help/latest/command/file.html#make-directory

**Comment by markdewing** [comment]
Design decision note for the history of this code:  Another way to solve this issue is to change how the "href" tags work, and make their file reference relative to the input file instead of the current directory (as it works now).   That would require capturing the path of the input file and passing it around to the parts of the code that execute the "href" tags. 
However, executing input files from somewhere other than the current directory is not a generally important enough use-case to make the effort to make a more general change now.  The change in this PR is good enough to address the problem for the performance tests.

**Review by ye-luo — CHANGES_REQUESTED** [review]
The need is clear. Create position-independent input file when feasible.
Please restore the change on J123.xml which doesn't contain href.

I will add change to make symlink to the source directory instead of build directory.
When QMC_SYMLINK_TEST_FILES=OFF, we keep the old behavior.

**Comment by ye-luo** [comment]
I heavily changed the PR @markdewing could you make a review pass?

**Review by markdewing — APPROVED** [review]

**Comment by ye-luo** [comment]
Test this please

**Pull request merged**